src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -480,5 +495,16 @@ internal static char GetCharsFromUtf32(uint codepoint, out char lowSurrogate)
                 return (char)((codepoint - 0x00010000) / 0x0400 + 0xD800);
             }
         }
+
+        public TestAccessor GetTestAccessor()
+            => new(this);
+
+        public readonly struct TestAccessor(SlidingTextWindow window)
+        {
+            private readonly SlidingTextWindow _window = window;
+
+            public void SetDefaultCharacterWindow()
+                => _window._characterWindow = new char[DefaultWindowLength];
```

Comment:
needed to do this as the backing char window for the lexer comes from a pool, and thus can see different length'ed arrays depending on if a larger array was returned to the pool at some point.  this allows the test to keep things isolated from other tests changing shared static state.